model/labelset.go

```diff
@@ -62,3 +63,21 @@ func (l LabelSet) MergeFromMetric(m Metric) {
 		l[k] = v
 	}
 }
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (l *LabelSet) UnmarshalJSON(b []byte) error {
+	var m map[LabelName]LabelValue
+	if err := json.Unmarshal(b, &m); err != nil {
+		return err
+	}
+	// encoding/json only unmarshals maps of the form map[string]T. It does not
+	// detect that LabelName is a string and does not call its UnmarshalJSON method.
```

Comment:
Isn't that misworded? You want to tell us that LabelName is treated as a string and not as  LabelName by encoding/json, don't you?